<launch>
  <!-- Arguments -->
  <arg name="robot_name" default="$(env ROBOT_NAME)"/>
  <arg name="model" default="waffle" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x" default="$(env X)"/>
  <arg name="y" default="$(env Y)"/>
  <arg name="z" default="0"/>
  <!--Robot namespace-->
  <group ns="$(arg robot_name)">
    <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />
    <!-- Spawn robot-->
    <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf"  args="-urdf -model $(arg robot_name) -x $(arg x) -y $(arg y) -z $(arg z) -param robot_description" />
    <!-- Load joint state controller -->
    <node pkg="joint_state_publisher" type="joint_state_publisher" name="joint_state_publisher">
      <param name="use_gui" value="true"/>
      <param name="rate" value="50"/>
      <param name="tf_prefix" value="$(arg robot_name)" />
    </node>
    <!-- Load robot state publisher -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
      <param name="publish_frequency" type="double" value="50.0" />
      <param name="tf_prefix" value="$(arg robot_name)" />
    </node>
    <!-- Load the controller -->
    <node pkg="multirobot_sim" type="simple_controller.py" name="simple_controller" output="screen" required="true">
      <param name="odom_topic" value="/$(arg robot_name)/odom"/>
      <param name="cmd_topic" value="/$(arg robot_name)/cmd_vel"/>
      <param name="goal_topic" value="/$(arg robot_name)/goal"/>
      <param name="path_topic" value="/$(arg robot_name)/path"/>
    </node>  
  </group>
</launch>
